src/Microsoft.AspNetCore.DataProtection/DataProtectionBuilderExtensions.cs

```diff
@@ -17,196 +17,195 @@
-    /// <see cref="ProtectKeysWithCertificate(string)" /> should generally be accompanied by
-    /// a call to <see cref="PersistKeysToFileSystem(DirectoryInfo)"/>, or exceptions may
-    /// occur at runtime due to the data protection system not knowing where to persist keys.
-    /// </para>
-    /// <para>
-    /// Similarly, when a developer modifies the default protected payload cryptographic
-    /// algorithms, it is intended that he also select an explitiy key storage location.
-    /// A call to <see cref="UseCryptographicAlgorithms(AuthenticatedEncryptionOptions)"/>
-    /// should therefore generally be paired with a call to <see cref="PersistKeysToFileSystem(DirectoryInfo)"/>,
-    /// for example.
-    /// </para>
-    /// <para>
-    /// When the default cryptographic algorithms or at-rest key protection mechanisms are
-    /// changed, they only affect <strong>new</strong> keys in the repository. The repository may
-    /// contain existing keys that use older algorithms or protection mechanisms.
-    /// </para>
-    /// </remarks>
-#else
-    /// <summary>
-    /// Provides access to configuration for the data protection system, which allows the
-    /// developer to configure default cryptographic algorithms, key storage locations,
-    /// and the mechanism by which keys are protected at rest.
-    /// </summary>
-    /// <remarks>
-    /// <para>
-    /// If the developer changes the at-rest key protection mechanism, it is intended that
-    /// he also change the key storage location, and vice versa.
-    /// </para>
-    /// <para>
-    /// Similarly, when a developer modifies the default protected payload cryptographic
-    /// algorithms, it is intended that he also select an explitiy key storage location.
-    /// A call to <see cref="UseCryptographicAlgorithms(AuthenticatedEncryptionOptions)"/>
-    /// should therefore generally be paired with a call to <see cref="PersistKeysToFileSystem(DirectoryInfo)"/>,
-    /// for example.
-    /// </para>
-    /// <para>
-    /// When the default cryptographic algorithms or at-rest key protection mechanisms are
-    /// changed, they only affect <strong>new</strong> keys in the repository. The repository may
-    /// contain existing keys that use older algorithms or protection mechanisms.
-    /// </para>
-    /// </remarks>
-#endif
-    public class DataProtectionConfiguration
+    public static class DataProtectionBuilderExtensions
     {
         /// <summary>
-        /// Creates a new configuration object linked to a <see cref="IServiceCollection"/>.
+        /// Sets the unique name of this application within the data protection system.
         /// </summary>
-        public DataProtectionConfiguration(IServiceCollection services)
+        /// <param name="builder">The <see cref="IDataProtectionBuilder"/>.</param>
+        /// <param name="applicationName">The application name.</param>
+        /// <returns>The 'this' instance.</returns>
```

Comment:
Looks like elsewhere in this repo we do the more correct "returns a reference to this instance after the operation completed" or whatever - so let's use that everywhere here.